src/index.test.mts

```diff
@@ -1618,36 +1619,32 @@ const runTests = (baseopts) => {
         const testPath = dpath('subdir/add.txt');
         const filename = sysPath.basename(testPath);
         options.cwd = sysPath.dirname(testPath);
-        await fs_mkdir(options.cwd);
+        await fsp.mkdir(options.cwd);
         await delay();
-        await write(testPath, 'hello');
+        await fsp.writeFile(testPath, 'hello');
         await delay();
         const spy = await aspy(cwatch(currentDir, options), EV.ALL);
-        await write(testPath, 'edit');
+        await fsp.writeFile(testPath, 'edit');
         await delay();
-        await fs_unlink(testPath);
+        await fsp.unlink(testPath);
         await waitFor([spy.withArgs(EV.UNLINK)]);
         spy.should.have.been.calledWith(EV.UNLINK, filename);
         spy.should.not.have.been.calledWith(EV.CHANGE, filename);
       });
-      describe('race condition', () => {
-        function w(fn, to) {
-          return setTimeout.bind(null, fn, to || slowerDelay || 50);
-        }
-        function simpleCb(err) {
-          if (err) throw err;
-        }
-
+      describe.skip('race condition', () => {
         // Reproduces bug https://github.com/paulmillr/chokidar/issues/546, which was causing an
         // uncaught exception. The race condition is likelier to happen when stat() is slow.
         const _realStat = fs.stat;
+        let statStub: sinon.SinonSpy;
+
         beforeEach(() => {
           options.awaitWriteFinish = { pollInterval: 50, stabilityThreshold: 50 };
           options.ignoreInitial = true;
 
           // Stub fs.stat() to take a while to return.
-          sinon.stub(fs, 'stat').callsFake((path, cb) => {
-            _realStat(path, w(cb, 250));
+          statStub = sinon.stub(fs, 'stat').callsFake(async (path, cb) => {
```

Comment:
this test can't have ever done what we thought since we moved to ESM here (i.e. since this became `test.mjs`)

in ESM you can't stub out a module's exports like this. which means it won't have been delaying the `fs.stat` calls

the test has been passing though... so maybe it just didn't actually test the thing it was trying to test? and passed because no such delay happened

Reply:
we can remove it